mirrord-layer/src/file.rs

```diff
@@ -106,3 +117,299 @@ impl OpenOptionsInternalExt for OpenOptionsInternal {
+        use FileResponse::*;
+        match message {
+            Open(open) => {
+                debug!("DaemonMessage::OpenFileResponse {open:#?}!");
+                pop_send(&mut self.open_queue, open)
+            }
+            Read(read) => {
+                // The debug message is too big if we just log it directly.
+                let file_response = read
+                    .inspect(|success| {
+                        debug!("DaemonMessage::ReadFileResponse {:#?}", success.read_amount)
+                    })
+                    .inspect_err(|fail| error!("DaemonMessage::ReadFileResponse {:#?}", fail));
+
+                pop_send(&mut self.read_queue, file_response)
+            }
+            Seek(seek) => {
+                debug!("DaemonMessage::SeekFileResponse {:#?}!", seek);
+                pop_send(&mut self.seek_queue, seek)
+            }
+            Write(write) => {
+                debug!("DaemonMessage::WriteFileResponse {:#?}!", write);
+                pop_send(&mut self.write_queue, write)
+            }
+            Close(close) => {
+                debug!("DaemonMessage::CloseFileResponse {:#?}!", close);
+                pop_send(&mut self.close_queue, close)
+            }
+        }
+    }
+
+    pub async fn handle_hook_message(
+        &mut self,
+        message: HookMessageFile,
+        codec: &mut actix_codec::Framed<
+            impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
+            ClientCodec,
+        >,
+    ) -> Result<(), LayerError> {
+        use HookMessageFile::*;
+        match message {
+            Open(open) => self.handle_hook_open(open, codec).await,
+            OpenRelative(open_relative) => {
+                self.handle_hook_open_relative(open_relative, codec).await
+            }
+
+            Read(read) => self.handle_hook_read(read, codec).await,
+            Seek(seek) => self.handle_hook_seek(seek, codec).await,
+            Write(write) => self.handle_hook_write(write, codec).await,
+            Close(close) => self.handle_hook_close(close, codec).await,
+        }
+    }
+
+    async fn handle_hook_open(
+        &mut self,
+        open: Open,
+        codec: &mut actix_codec::Framed<
+            impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
+            ClientCodec,
+        >,
+    ) -> Result<(), LayerError> {
+        let Open {
+            file_channel_tx,
+            path,
+            open_options,
+        } = open;
+        debug!(
+            "HookMessage::OpenFileHook path {:#?} | options {:#?}",
+            path, open_options
+        );
+
+        self.open_queue.push_back(file_channel_tx);
+
+        let open_file_request = OpenFileRequest { path, open_options };
+
+        let request = ClientMessage::FileRequest(FileRequest::Open(open_file_request));
+        codec.send(request).await.map_err(From::from)
```

Comment:
Why do you need most of these `map_err(From::from)`? Can't you do `Ok(...?)` instead?

```rust
Ok(codec.send(request).await?)
```

Reply:
The map felt more cool then wrapping it in an Ok, guess it's a style-decision..